sdk/search/search-documents/review/search-documents.api.md

```diff
@@ -980,28 +1418,27 @@ export interface PatternTokenizer {
 }
 
 // @public
-export type PhoneticEncoder = 'metaphone' | 'doubleMetaphone' | 'soundex' | 'refinedSoundex' | 'caverphone1' | 'caverphone2' | 'cologne' | 'nysiis' | 'koelnerPhonetik' | 'haasePhonetik' | 'beiderMorse';
+export type PhoneticEncoder = "metaphone" | "doubleMetaphone" | "soundex" | "refinedSoundex" | "caverphone1" | "caverphone2" | "cologne" | "nysiis" | "koelnerPhonetik" | "haasePhonetik" | "beiderMorse";
 
 // @public
-export interface PhoneticTokenFilter {
-    encoder?: PhoneticEncoder;
-    name: string;
+export type PhoneticTokenFilter = TokenFilterBase & {
     odatatype: "#Microsoft.Azure.Search.PhoneticTokenFilter";
+    encoder?: PhoneticEncoder;
     replaceOriginalTokens?: boolean;
-}
+};
 
 // @public
-export type QueryType = 'simple' | 'full';
+export type QueryType = "simple" | "full";
 
 // @public
-export type RegexFlags = 'CANON_EQ' | 'CASE_INSENSITIVE' | 'COMMENTS' | 'DOTALL' | 'LITERAL' | 'MULTILINE' | 'UNICODE_CASE' | 'UNIX_LINES';
+export type RegexFlags = string;
 
 // @public
 export type ResetIndexerOptions = OperationOptions;
 
 // @public
 export interface ResourceCounter {
-    quota?: number;
+    quota?: number | null;
```

Comment:
what's the deal with these things being optional or null? What is the service distinction between `null` and `undefined` ?

Reply:
Provided a detailed explanation in a another comment.